Content.Server/CartridgeLoader/Cartridges/MessagesCartridgeSystem.cs

```diff
@@ -0,0 +1,310 @@
+                continue;
+
+            if (transform.MapID == mapId &&
+                power.Powered)
+            {
+                //_sawmill.Debug("Server found");
+                return (uid, messageServer);
+            }
+        }
+       // _sawmill.Debug("No server");
+        return null;
+    }
+
+    private string GetName(MessagesCartridgeComponent component, string key)
+    {
+        if (!(component.NameDict.ContainsKey(key)))
+        {
+            return "Unknown";
+        }
+        else
+        {
+            if (component.NameDict[key][0] == '(')
+            {
+                return "Unknown "+component.NameDict[key];
+            }
+            else
+            {
+                return component.NameDict[key];
+            }
+        }
+    }
+
+    private void UpdateUiState(EntityUid uid, EntityUid loaderUid, MessagesCartridgeComponent? component)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+        MessagesUiState state;
+        if (component.ChatUid == null)
+        {
+            List <(string,string)> userList = new();
+
+            foreach (var nameEntry in component.NameDict.Keys)
+            {
+                if (nameEntry == component.UserUid)
+                    continue;
+                userList.Add((GetName(component,nameEntry),nameEntry));
+            }
+
+            userList.Sort(delegate((string,string) a, (string,string) b)
+            {
+                    return String.Compare(a.Item2,b.Item2);
+            });
+
+            state = new MessagesUiState(MessagesUiStateMode.UserList, userList, null);
+        }
+        else
+        {
+            List<MessagesMessageData> messageList = new();
+
+            foreach (var message in component.Messages)
+            {
+                if ((message.SenderId == component.ChatUid && message.ReceiverId == component.UserUid) || (message.ReceiverId == component.ChatUid && message.SenderId == component.UserUid))
+                {
+                    messageList.Add(message);
+                }
+            }
+
+            messageList.Sort(delegate(MessagesMessageData a, MessagesMessageData b)
+            {
+                return TimeSpan.Compare(a.Time,b.Time);
+            });
+
+            List<(string, string)> formattedMessageList = new();
+
+            foreach (var message in messageList)
+            {
+                string name = GetName(component, message.SenderId);
+                var stationTime =message.Time.Subtract(_gameTicker.RoundStartTimeSpan);
+                string content = stationTime.ToString("\\[hh\\:mm\\:ss\\]")+" : "+message.Content;
+                formattedMessageList.Add((name , content));
+            }
+
+            state = new MessagesUiState(MessagesUiStateMode.Chat, formattedMessageList, GetName(component, component.ChatUid));
+        }
+        _cartridgeLoaderSystem?.UpdateCartridgeUiState(loaderUid, state);
+    }
+
```

Comment:
As mentioned above configure your IDE to provide warnings because there's a bunch of dummy code here such as the null operator that is unnecessary.